common/src/main/java/com/hedera/block/common/utils/FileUtils.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.common.utils;
+
+import com.hedera.block.common.constants.ErrorMessageConstants;
+import com.hedera.block.common.constants.StringConstants;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.System.Logger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileAttribute;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.Objects;
+import java.util.Set;
+import java.util.zip.GZIPInputStream;
+
+/** A utility class that deals with logic related to dealing with files. */
+public final class FileUtils {
+    private static final Logger LOGGER = System.getLogger(FileUtils.class.getName());
+
+    /**
+     * Default file permissions defines the file and directory for the storage package.
+     *
+     * <p>Default permissions are set to: rwxr-xr-x
+     */
+    @NonNull
```

Comment:
Again, why `@NonNull` I think we should only use this decorator on parameters.

Reply:
Again was in the original code, I agree to remove it.

Reply:
removed